web/src/store/index.ts

```diff
@@ -433,7 +442,7 @@ const actions = {
     }
   },
 
-  async [SET_DATASET_DESCRIPTION]({ commit }, { dataset_id, description }) {
+  async [SET_DATASET_DESCRIPTION]({ commit }: { commit: any }, {dataset_id, description}: { dataset_id: string, description: string }) {
```

Comment:
You've done `{ commit } : any` above and something a little different here.

The people in [this article](https://codeburst.io/vuex-and-typescript-3427ba78cfa8) didn't annotate the first argument to vuex methods at all. What are the benefits of each?

Reply:
The `any`s in both instances are placeholders and therefore equally "bad" (i.e., must be replaced eventually by an actual type). `{ commit }: any` is "more equal" because it's less precise. But in this case, both type declarations have the exact same semantics.

So the benefit of the latter approach is that it already has the correct shape for the eventually correct typing for that expression. Better even than that is probably to do something like

```typescript
interface CommitArg {
  commit: ...;
}
```

and then type the expression as `{ commit }: CommitArg`.

I assume that article didn't type the first argument either because they were focused on demonstrating something else, or the TS compiler was able to determine the type of that argument solely from usage in their example codebase. I stuck in a placeholder typing because the compiler told me it couldn't infer the type. (Though, honestly, I believe it's better to type as much as possible in order to avoid problems with type-compatible but semantically forbidden usages later in the lifecycle of the codebase.)